Simplenote/SPNavigationController.swift

```diff
@@ -120,45 +113,56 @@ class SPNavigationController: NSViewController {
         viewStack.append(child)
     }
 
-    @discardableResult
-    private func attachView(subview: NSView, below siblingView: NSView?) -> (leading: NSLayoutConstraint, trailing: NSLayoutConstraint)? {
-        if let siblingView {
+    private func attachView(subview: NSView, below siblingView: NSView?, animated: Bool) {
+
+        let padding = Constants.buttonViewLeadingPadding + Constants.buttonViewHeight
+        let finalHeight = subview.fittingSize.height + padding
+
+        if let siblingView,
+           animated {
+            heightConstraint?.constant = siblingView.fittingSize.height + padding
             view.addSubview(subview, positioned: .below, relativeTo: siblingView)
         } else {
             view.addSubview(subview)
         }
 
         subview.translatesAutoresizingMaskIntoConstraints = false
 
-        let leadingAnchor = subview.leadingAnchor.constraint(equalTo: view.leadingAnchor)
-        let trailingAnchor = subview.trailingAnchor.constraint(equalTo: view.trailingAnchor)
-
         NSLayoutConstraint.activate([
-            leadingAnchor,
-            trailingAnchor,
-            subview.topAnchor.constraint(equalTo: backButton.bottomAnchor),
-            subview.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+            subview.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            subview.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            subview.topAnchor.constraint(equalTo: backButton.bottomAnchor)
         ])
 
+        guard animated else {
+            heightConstraint?.constant = finalHeight
+            return
+        }
+
+        NSAnimationContext.runAnimationGroup { context in
+            context.duration = 0.4
+            context.timingFunction = .init(name: .easeInEaseOut)
+
+            heightConstraint?.animator().constant = finalHeight
+        }
```

Comment:
Do you think we could decouple `attach` + `resizeWindow` ?

And then, if we needed, have a third API that calls them both?

```
func attachView(subview: NSView, below siblingView: NSView?, animated: Bool) 
func resizeWindowHeight(for subview: NSView, animated: Bool
```

And then...

```
func attachViewAndResizeWindow(subview: NSView, below siblingView: NSView?, animated: Bool) 
```

Reply:
**Reason** why this caught my eye is: the `animation` is only applicable to the `Window Resize`, not to the View we're pushing, right?

**ALSO** there's the `animateTransition` API, so... you can `attach(..., animated: true)`, but it won't animate the transition, just the Window Resize